source/Octopus.Tentacle/Scripts/PowerShell.cs

```diff
@@ -1,59 +1,24 @@
 ﻿using System;
-using System.IO;
-using System.Text;
 
 namespace Octopus.Tentacle.Scripts
 {
     public class PowerShell : IShell
     {
-        const string EnvPowerShellPath = "PowerShell.exe";
-        static string? powerShellPath;
-
+        readonly IShell inner;
         public string Name => nameof(PowerShell);
 
-        public string GetFullPath()
-            => GetFullPowerShellPath();
-
-        public static string GetFullPowerShellPath()
+        public PowerShell()
         {
-            if (powerShellPath != null)
-                return powerShellPath;
-
-            try
-            {
-                var systemFolder = Environment.GetFolderPath(Environment.SpecialFolder.System);
-                powerShellPath = Path.Combine(systemFolder, @"WindowsPowershell\v1.0\", EnvPowerShellPath);
-
-                if (!File.Exists(powerShellPath))
-                    powerShellPath = EnvPowerShellPath;
-            }
-            catch (Exception)
-            {
-                powerShellPath = EnvPowerShellPath;
-            }
-
-            return powerShellPath;
+            var powerShellCore = new PowerShellCore();
+            if (powerShellCore.PowerShellCoreExists)
```

Comment:
If we were to do this, we would probably need some sort of configuration to disable/enable.

Also, is it sufficient to find PowerShell core? In the change for the hardening script the branching logic seems to only switch for PowerShell core >= 7.3. Do we need to know the version of PowerShell core?

Reply:
The 7.3 check is to work around a change they made to not load the security stuff - https://github.com/PowerShell/PowerShell/issues/18896

Ironically, it was designed to speed up powershell :)